.github/workflows/linux.yml

```diff
@@ -54,6 +54,11 @@ jobs:
             std: 20
             cxxflags: -stdlib=libc++
             install: sudo apt install libc++-11-dev libc++abi-11-dev
+          - cxx: clang++-16
+            build_type: Debug
+            std: 20
+            cxxflags: -stdlib=libc++
+            install: sudo apt install g++-11 libstdc++-11-dev clang-16 libomp-16-dev libc++-16-dev libc++abi-16-dev
```

Comment:
We have too many CI configs already and shouldn't introduce yet another one for this.

Reply:
Maybe be better to remove the older ``clang``?
For example ``clang++-8``?

Reply:
The old one is more likely to catch compatibility issues.

Reply:
But old don't catch compatibility issues with new libc++.
I think we need CI for the newest versions of the all compilers, which is updated periodically.
If you tell me to delete this CI (or update to lasted clang-17), I will do so.

Reply:
We care about old compilers because of explicit compatibility guarantees which are easy to accidentally break and need to be tested against. The newest versions on the other hand rarely cause real problems because they usually ensure compatibility with existing code. We should care about commonly used and not latest versions. It is a bit hard to estimate which versions are more used but I think a reasonable rule of thumb is to have a year or two lag or check which versions ship with popular LTS Linux releases. I agree that we need to periodically bump versions (but not to the latest). The only exception is when we need to test against specific major features like modules or consteval. This particular case is just a warning in test which doesn't need extra coverage so please remove the new CI config.

Reply:
Done.